torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -894,7 +932,12 @@ class FullyShardedDataParallel(nn.Module):
             This can also help load checkpoints taken by ``state_dict`` and to be loaded by
             ``load_state_dict`` in a memory efficient way. See documentation for
             :class:`FullStateDictConfig` for an example of this. (Default: ``False``)
-
+        limit_all_gathers (bool): If ``False``, then FSDP allows the CPU
+            thread to schedule all-gathers without any extra synchronization.
+            If ``True``, then FSDP explicitly synchronizes the CPU thread to
+            prevent too many in-flight all-gathers. This ``bool`` only affects
```

Comment:
should we note that this is 2 at the moment? Do we want to expose any knob for further tuning by advanced users?

Reply:
At the moment, I do not want to expose the knob (see the updated PR summary). Given that we do not expose the knob, I am not sure if it is valuable to explain the internal limiter implementation to the user.